### DISCUSSION TITLE
Support for PRF extension for Passkeys

### DISCUSSION THREAD

**Discussion by mrinalwahal** [zitadel/zitadel#7507] [Q&A]
Hey,

I understand that Zitadel has very good support for passkeys. As detailed [over here](https://zitadel.com/docs/guides/integrate/login-ui/passkey).

For my specific use case, I'm building a custom password manager. And I wanted to inquire about the support for [PRF extension for WebAuthN](https://github.com/w3c/webauthn/wiki/Explainer:-PRF-extension) which essentially allows us to generate a unique secret key from the passkey, and enable us to use the passkey for general purpose encryption/decryption. This feature can allow me to encrypt and decrypt my users' data directly on client side only using passkeys.

Is the support for [WebAuthn PRF extension](https://bitwarden.com/blog/prf-webauthn-and-its-role-in-passkeys/) already available or is it part of the roadmap?

This is a make or break decision for me regarding using Zitadel for my project.

**Comment by hifabienne** [comment]
@livio-a @muhlemmer Can you help here?

**Comment by muhlemmer** [comment]
No, we currently do not support PRF extenstion. Your use-case is very specific and might not be useful to the majority of zitadel users.  

We use the [go-webauthn](https://github.com/go-webauthn/webauthn) library for passkeys. They already have an issue open for extension support: https://github.com/go-webauthn/webauthn/issues/123. If it's very important for your project, according regular FOSS convention I would recommend contributing to the feature there. Once it is available we can see how we make it available in zitadel.

If your project is commercial you can also have a look at our [Feature Development Policy](https://zitadel.com/docs/legal/policies/feature-development-policy) to speed up implementation, which might include us doing the contribution to go-webauthn.
